client/matching/weighted_loadbalancer.go

```diff
@@ -133,61 +131,45 @@ func NewWeightedLoadBalancer(
 }
 
 func (lb *weightedLoadBalancer) PickWritePartition(
-	domainID string,
-	taskList types.TaskList,
 	taskListType int,
-	forwardedFrom string,
+	req WriteRequest,
 ) string {
-	return lb.fallbackLoadBalancer.PickWritePartition(domainID, taskList, taskListType, forwardedFrom)
+	return lb.fallbackLoadBalancer.PickWritePartition(taskListType, req)
 }
 
 func (lb *weightedLoadBalancer) PickReadPartition(
-	domainID string,
-	taskList types.TaskList,
 	taskListType int,
-	forwardedFrom string,
+	req ReadRequest,
+	isolationGroup string,
 ) string {
-	if forwardedFrom != "" || taskList.GetKind() == types.TaskListKindSticky {
-		return taskList.GetName()
-	}
-	if strings.HasPrefix(taskList.GetName(), common.ReservedTaskListPrefix) {
-		return taskList.GetName()
-	}
 	taskListKey := key{
-		domainID:     domainID,
-		taskListName: taskList.GetName(),
+		domainID:     req.GetDomainUUID(),
+		taskListName: req.GetTaskList().GetName(),
 		taskListType: taskListType,
 	}
 	wI := lb.weightCache.Get(taskListKey)
 	if wI == nil {
-		return lb.fallbackLoadBalancer.PickReadPartition(domainID, taskList, taskListType, forwardedFrom)
+		return lb.fallbackLoadBalancer.PickReadPartition(taskListType, req, isolationGroup)
```

Comment:
`multiLoadBalancer` has the common checks and must be the only one used directly as wrapper going forward. It also has a `defaultLoadBalancer`. So we may simplify `weightedLoadBalancer` by getting rid of fallback logic and returning a special error to indicate fallback to default lb is needed.

Reply:
Agreed. Going to address in followup refactor.